File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size()==1) return nums[0];
        int ans=INT_MIN;
        for(size_t i=0;i<nums.size();i++){
            int count=0;
            for(size_t j=i;j<nums.size();j++){
                count+=nums[j];
                ans=max(count,ans);
            }
        }
        return ans;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let ans=nums[0]
    for(let i=1;i<nums.length;i++){
        nums[i]=Math.max(nums[i],nums[i]+nums[i-1])
        ans=Math.max(ans,nums[i])
    }
    return ans
};